src/attributes/classes.js

```diff
@@ -24,19 +22,17 @@ jQuery.fn.extend({
 			});
 		}
 
-		if ( proceed ) {
-			// The disjunction here is for better compressibility (see removeClass)
-			classes = ( value || "" ).match( rnotwhite ) || [];
+		if ( typeof value === "string" && value ) {
+			classes = value.match( rnotwhite ) || [];
 
-			for ( ; i < len; i++ ) {
-				elem = this[ i ];
+			while ( ( elem = this[ i++ ] ) ) {
```

Comment:
I don't think falsy values cause issues, there are a lot of places where Sizzle iterates node lists and uses this construct. I like the `while` better than the headless-`for` as a reader, so there's that. :smile: